test/Error.spec.js

```diff
@@ -7,8 +7,8 @@ const expect = require('chai').expect;
 
 describe('Error', () => {
   it('Populates with type and message params', () => {
-    const e = new Error('FooError', 'Foo happened');
-    expect(e).to.have.property('type', 'FooError');
+    const e = new Error.GenericError('Foo happened');
+    expect(e).to.have.property('type', 'GenericError');
```

Comment:
Can this test change be reverted? I think this test should continue to pass.

Reply:
Ahhh, I see, this is what you were talking about. The Error exported from the root has a totally different signature from `GenericError`, and also doesn't make much sense to extend... hmm...

Reply:
We should probably keep all the code for `function _Error(raw) {` and just mark it as deprecated (and not actually inherit from it in the ES6 classes). Thoughts?

Reply:
`Error` here is default import from `Error.js`. To make it pass we should be doing weird things as I commented earlier, like this:

```js
module.exports = GenericError;
module.exports.StripeError = StripeError;
```

I mean, export `GenericError` with other errors as attached properties on it...
The example you put makes sense, `extend` export do work, but doing this will look waoh...

Reply:
I'm thinking this: 
```
module.exports = _Error;

/**
 * DEPRECATED
 * Generic Error klass to wrap any errors returned by stripe-node
 */
function _Error(raw) {
  this.populate(...arguments);
  this.stack = new Error(this.message).stack;
}

// Extend Native Error
_Error.prototype = Object.create(Error.prototype);

_Error.prototype.type = 'GenericError';
_Error.prototype.populate = function(type, message) {
  this.type = type;
  this.message = message;
};

_Error.extend = utils.protoExtend;

class StripeError {...}
class ...
module.exports.StripeError = StripeError;
...
```

Reply:
Weeeell... Can't do THAT ugly. Passing all original tests now ✅